packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -437,7 +437,8 @@ class MethodChannel {
     } on MissingPluginException {
       return null;
     } catch (e) {
-      return codec.encodeErrorEnvelope(code: 'error', message: e.toString(), details: null);
+      codec.encodeErrorEnvelope(code: 'error', message: e.toString(), details: null);
+      rethrow;
```

Comment:
@goderbauer I could use a quick preliminary review from you when you have time.

What do you think about this solution?  Any idea if we were purposely swallowing all of these errors here for some reason, or what the intention was?

Also, if this is a valid solution, why isn't my test below able to test this exception?

Reply:
We are not really swallowing them. We are sending the error back to the native side, so it can be dealt with there, is my understanding.

Why do you need to rethrow the error here? And aren't these just the errors of the user-provided `handler`? If you really want to catch them, could you just have a try-catch in your `handler`?

Reply:
I just wrote this PR as a quick exploration of #65980, so I don't have a strong opinion, but I'd argue that:

  * Users don't want or expect errors that happen in their callback code to be hidden from them.
  * Updating the docs to inform users that they need to wrap all action callbacks in a try/catch if they want to be informed of errors is a possibility, but that's not an ideal developer experience, and I expect that users would continue to be confused by this behavior.
  * I'm not familiar with this code or error handling in the engine, but I would think that the Flutter framework wants to be informed of its own errors that happen as a result of these messages.

Or maybe I don't understand this problem fully. Are the errors that get sent to the engine logged somewhere that the user has access to? Maybe it's different in production mode?

Reply:
I see, thanks for the additional context. Indeed, it doesn't make sense to send errors that happen during the user-provided `onSubmit` back to the engine. Can we just wrap the location where we call `onSubmit` in a try-catch and then report a FlutterError from the catch block? E.g. here:

https://github.com/flutter/flutter/blob/e45157fc08c90ab896988f61ea7c8cb95c95f96e/packages/flutter/lib/src/widgets/editable_text.dart#L1825-L1826

We use this pattern in other places as well where we call user-provided callbacks, example:

https://github.com/flutter/flutter/blob/e45157fc08c90ab896988f61ea7c8cb95c95f96e/packages/flutter/lib/src/animation/listener_helpers.dart#L134-L145

Reply:
Yes thanks for the tip, I'll do that!  I was unaware we already were using this sort of pattern.  Will mention you when I'm done.